pulp_ansible/app/galaxy/v3/filters.py

```diff
@@ -0,0 +1,174 @@
+from django.contrib.postgres.search import SearchQuery
+from django.db.models import fields as db_fields
+from django.db.models import Q
+from django.db.models.expressions import F, Func
+from django_filters import (
+    filters,
+    FilterSet,
+    OrderingFilter,
+)
+import semantic_version
+from rest_framework.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from pulpcore.plugin.viewsets import LabelFilter
+
+from pulp_ansible.app import models
+
+
+class CollectionVersionSearchFilter(FilterSet):
+    """A custom filterset for cross-repo search."""
+
+    strict = False
+
+    name = filters.CharFilter(field_name="collection_version__name", lookup_expr="exact")
+    namespace = filters.CharFilter(field_name="collection_version__namespace", lookup_expr="exact")
+    version = filters.CharFilter(field_name="collection_version__version", lookup_expr="exact")
+    repository_name = filters.CharFilter(method="filter_by_repository_name")
+    repository = filters.CharFilter(method="filter_by_repository_id")
+    is_highest = filters.BooleanFilter(field_name="is_highest")
+    highest = filters.BooleanFilter(field_name="is_highest")
+    is_deprecated = filters.BooleanFilter(field_name="is_deprecated")
+    deprecated = filters.BooleanFilter(field_name="is_deprecated")
+    is_signed = filters.BooleanFilter(field_name="is_signed")
+    signed = filters.BooleanFilter(field_name="is_signed")
+    q = filters.CharFilter(field_name="q", method="filter_by_q")
+    tags = filters.CharFilter(
+        field_name="tags",
+        method="filter_by_tags",
+        help_text="Filter by comma separate list of tags that must all be matched",
+    )
+    distribution = filters.CharFilter(method="filter_by_distribution_id")
+    distribution_base_path = filters.CharFilter(method="filter_by_base_path")
+    dependency = filters.CharFilter(method="filter_by_dependency")
+    version_range = filters.CharFilter(field_name="version_range", method="version_range_filter")
+    repository_version = filters.CharFilter(
+        field_name="repository_version__number", method="repository_version_filter"
+    )
+    keywords = filters.CharFilter(field_name="q", method="filter_by_q")
+    repository_label = LabelFilter(label_field_name="repository__pulp_labels")
```

Comment:
I don't think this is working, using a filter like `?repository_label=approved`, but the `pulp_labels` are in the format :arrow_down: 
```
pulp_labels: {
   pipeline: "approved"
}
```

Reply:
okay, found out it can be used like `?repository_label=pipeline=approved`, my bad :facepalm:

Reply:
Yep, just make sure you encode that string.